src/Models/HydrostaticFreeSurfaceModels/pcg_implicit_free_surface_solver.jl

```diff
@@ -132,7 +132,8 @@ function implicit_free_surface_linear_operation!(L_ηⁿ⁺¹, ηⁿ⁺¹, ∫
     grid = L_ηⁿ⁺¹.grid
     arch = architecture(L_ηⁿ⁺¹)
 
-    # REMEMBER!!! This is going to create problems!!!!
+    # REMEMBER: This is going to create problems for multi-region! 
+    #           (Non-local operation cannot be done within @apply_regionally)
```

Comment:
```suggestion
    # Note: because of `fill_halo_regions!` below, we cannot use `PCGImplicitFreeSurface` on a multi-region grid
    # (`fill_halo_regions!` cannot be used within `@apply_regionally`)
```